srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditVariable.java

```diff
@@ -0,0 +1,219 @@
+package gov.va.med.srcalc.web.view.admin;
+
+import java.util.*;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSortedSet;
+
+import gov.va.med.srcalc.domain.model.*;
+import gov.va.med.srcalc.service.ModelInspectionService;
+
+/**
+ * Stores basic {@link AbstractVariable} properties for creating a new or
+ * editing an existing variable.
+ */
+public abstract class EditVariable
+{
+    private final ImmutableSortedSet<VariableGroup> fAllGroups;
+    
+    private String fKey;
+
+    private String fDisplayName;
+    
+    private Optional<String> fHelpText;
+    
+    private int fGroupId;
+    
+    private final SortedSet<RiskModel> fDependentModels;
+    
+    /**
+     * Constructs an instance with default values for all properties.
+     * @param modelService to provide reference data (e.g., available
+     * VariableGroups) to the user
+     */
+    public EditVariable(
+            final ModelInspectionService modelService) 
+    {
+        // Sort the groups per the getAllGroups() contract.
+        fAllGroups = ImmutableSortedSet.copyOf(modelService.getAllVariableGroups());
+        fKey = "";
+        fDisplayName = "";
+        fHelpText = Optional.absent();
+        fGroupId = 0;
+        fDependentModels = new TreeSet<>();
+    }
+    
+    /**
+     * Constructs an instance with the properties initialized based on the given
+     * existing variable.
+     * @param variable an existing variable containing the initial properties.
+     * Not stored.
+     * @param modelService to provide reference data (e.g., available
+     * VariableGroups) to the user
+     */
+    public EditVariable(
+            final AbstractVariable variable,
+            final ModelInspectionService modelService)
+    {
+        this(modelService);
```

Comment:
Why bother with the constructor when the only thing that is really necessary is to get the variable groups and create a TreeSet?